internal/worker/dbaccessor/worker_integration_test.go

```diff
@@ -128,6 +130,16 @@ func (s *integrationSuite) TearDownTest(c *gc.C) {
 	if dqlite.Enabled && s.worker != nil {
 		workertest.CleanKill(c, s.worker)
 	}
+	if s.db != nil {
+		s.db.Close()
+	}
+
+	// We can't for what ever reason call s.DqliteSuite.TearDownTest here
+	// because it will cause a segfault. The workaround is to close the
+	// db directly.
+	if db := s.DB(); db != nil {
+		db.Close()
+	}
```

Comment:
The segfault in question: https://pastebin.canonical.com/p/Bn6rf3dGxh/
We're jumping over certain suites to prevent default initialization and that's causing nil pointers.